### PR TITLE
Disable Create when over quota, remove Beta/Free copy

### PR DESCRIPTION
**Pull request by pbrisbin**
The landing page is a little sparse now, but I'm not the best copy writer...

### CODE REVIEW

File: Handler/Sites.hs [hunk 1]
```diff
@@ -31,6 +32,7 @@ postSitesR = do
         setMessage "Site created!"
         redirect $ SiteR siteId
 
+    plan <- runDB $ belongsToJust userPlan user
```
Comment (jferris) [inline Handler/Sites.hs]: This is cool.

File: Handler/Sites.hs [hunk 2]
```diff
@@ -22,7 +23,7 @@ getSitesR = do
 
 postSitesR :: Handler Html
 postSitesR = do
-    userId <- requireAuthId
+    Entity userId user <- requireAuth
```
Comment (jferris) [inline Handler/Sites.hs]: Do we want to add a quota check here to prevent race conditions or people hacking the form?

Comment (pbrisbin) [inline Handler/Sites.hs]: I don't care about people hacking the form, myself. The effort required to craft a request (with an authenticated session) is enough of a deterrent IMO. Could be worth, at some point, tracking if enough people are curiously over quota to warrant locking down.

If I understand it correctly, there's no race condition now -- but there would be, if we added the check here. Is that right?

Comment (jferris) [inline Handler/Sites.hs]: If you submit the form from two different sessions at once, you'll create two sites and the button will never be disabled.

Comment (jferris) [inline Handler/Sites.hs]: > The effort required to craft a request

I would personally just use the web inspector to enable the button.

Comment (pbrisbin) [inline Handler/Sites.hs]: Ha true.

Comment (pbrisbin) [inline Handler/Sites.hs]: Done.

File: Model/Site.hs [hunk 3]
```diff
@@ -38,3 +39,6 @@ destroySite siteId = do
     deleteWhere [CommentSite ==. siteId]
     deleteWhere [MembershipSite ==. siteId]
     delete siteId
+
+overSiteQuota :: Int -> Plan -> Bool
```
Comment (jferris) [inline Model/Site.hs]: Looks like this is always called with `length sites` - thoughts on changing the type to `[a] -> Plan Bool`?

Comment (pbrisbin) [inline Model/Site.hs]: Sure, why not.

**Comment by jferris** [comment]
I had a suggestion about the `overSiteQuota` function, but I think this is ready to merge.

I think there's a lot we can do to improve our test factories, although I think this isn't the place. For example, if `createSite` could generate valid, unique sites, you could just use `replaceM_`.

**Pull request merged**